### ISSUE TITLE
Temporarily lost acces to package while transfering to organisation.

### ISSUE THREAD

**Issue by Carreau** [pypi/warehouse#13558]
**Describe the bug**

 - I am (was) an owner of `IPython` project on Pypi. 
 - I am not an owner of pypi `Jupyter` team
 - I am a manager of the Pypi `Jupyter`/`IPython` team. 
 - I want to transfer IPython to the `Jupyter` org under the `IPython` team (to keep access). 
 - I transfer to `Jupyter` organisation, expecting to be prompted for a team. 
   - I lose access to manage the `IPython` package or assign it a team.
 - I need to reach to a pypi `Jupyter` owner to assign the `IPython` project it to the `IPython` team and regain access. 

Things that are unclear. 
 - I got remove from the list of collaborator (https://pypi.org/manage/project/ipython/collaboration/) but others IPython Project Owners (Fernando Perez, Brian granger... etc) and kept access, that feels weird. Why was I removed ? Because I was the transferer ? I would expect either everyone, or no-one to be removed. 

**Expected behavior**

Either:
 - Only allow transfer of project if transferer is owner on both source and destination. 
 - Allow to assign a team directly when transfering to keep access. 
 - Big warning "You will lose access to this project" with confirm dialog. 

**To Reproduce**

See Bug above. 

**My Platform**

NA

**Additional context**

Thanks **a lot** for organisations, I'm sure the UI/UX will be refined in the future. I do not exclude a Problem Exists Between Keyboard and Chair, but there was 4 people on a zoom call and we were all confused. This is also trying to gather feedback while still not used to the feature as I know new user feedback can be hard to get. 

Thanks.

**Comment by di** [comment]
Thanks for the issue. This is https://github.com/pypi/warehouse/issues/12423, I think we need to revisit that given the confusion here.

**Comment by Carreau** [comment]
Ah sorry missed it, I looked only in open issues. Feel free to close. 

Much love again.

**Comment by ewdurbin** [comment]
This sounds like a bug. I don't believe you should have been able to perform the transfer without being an Owner of the `Jupyter` org.

**Comment by webknjaz** [comment]
> without being an Owner of the `Jupyter` org.

Manager access is enough currently. I've faced this in the past with project mass transfers — an org manager can transfer a project where they are an owner and then, they're removed from that project.
I'd expect the owner to be preserved in addition to moving under the org. This is not only problematic because of the access loss but also, the person is no longer displayed as a maintainer/collaborator on the public project page, which many people would likely want.

**Comment by di** [comment]
This would be resolved by #14455. I think we should revisit that given the amount of user confusion that's happening here.

**Comment by d33bs** [comment]
Thanks @di ! I noticed the PR you cited was closed. Just my 2 cents: the scenario you brought up in the PR and which I mentioned here seem like a bug; a project owner shouldn't lose access if they add the project to an org they have membership to.

**Comment by di** [comment]
I tend to agree but the overall intent for how project and organization roles work means this is not a bug, see https://github.com/pypi/warehouse/issues/18102 for more details.

**Comment by di** [comment]
Closing as a duplicate of #12423, see #18102 for more details.
